mlflow/models/container/__init__.py

```diff
@@ -187,13 +188,22 @@ def _serve_pyfunc(model, env_manager):
         # Since MLServer will run without NGINX, expose the server in the `8080`
         # port, which is the assumed "public" port.
         port = DEFAULT_MLSERVER_PORT
+        model_dict = model.to_dict()
+        inference_server_kwargs = {
+            "model_name": model_dict.get("model_name"),
+            "model_version": model_dict.get(
+                "model_version", model_dict.get("run_id", model_dict.get("model_uuid"))
+            ),
+        }
     else:
         inference_server = scoring_server
         # users can use GUNICORN_CMD_ARGS="--workers=3" var env to override the number of workers
         nworkers = cpu_count
         port = DEFAULT_INFERENCE_SERVER_PORT
 
-    cmd, cmd_env = inference_server.get_cmd(model_uri=MODEL_PATH, nworkers=nworkers, port=port)
+    cmd, cmd_env = inference_server.get_cmd(
+        model_uri=MODEL_PATH, nworkers=nworkers, port=port, **inference_server_kwargs
```

Comment:
What happens when this additional metadata is passed to an mlserver instance of an older version of MLflow? This will throw a TypeError, right?

Reply:
Hi @BenWilson2 , 
Thank you for your feed back !

But I don't understand your point here, because `inference_server.get_cmd` is not part of MLServer lib but MLFlow lib itself (see https://github.com/M4nouel/mlflow/commit/59f85f48830921a0ea2c1ed6285c0b5b56bb7d97#diff-b9a92aa59c788dc99a7dc0bb6fee0c578603c2848768b8bc51b35e4c20cb3098L12).
And, in `inference_server.get_cmd`, `model_name` and `model_version` are passed to MLServer through environment variables, therefore there is no riks of raising error...

Did I misunderstand something ?

Reply:
Ah yes, this won't be an issue. Stay tuned, we're going to discuss internally if there are any integrations that we have that may be affected by this change (I can't imagine that there will be, but we're going to check just in case :) )

Reply:
Hi @BenWilson2 ,

Have you been able to check whether this PR affects anything else?
Can I help you in any way? Or suggest a different implementation that would better suit this project?
(My MLOps colleagues are looking forward to this feature :D)